### PR TITLE
Added templating and configuring simulations to the how it works page

### PR DESCRIPTION
**Pull request by maiyahershey**
This PR is for adding the missing "**templating mechanism**" and "**configuring simulations**" to the list on how the API works page and makes sure text is uniform throughout.

### CODE REVIEW

File: docs/explore_api/how_it_works.md [hunk 1]
```diff
@@ -26,23 +26,30 @@ nuances and additional options available, which we'll explore in detail througho
...
-- Configuring Simulations:
+- [Tasks](./tasks): Get to know the API's core computational object which gets created
+when you submit your simulation request, enabling you to track its progress and 
+access outputs in real-time.
+
+- [Shared and Dedicated Resources](./shared_dedicated_resources.md): Discover how
+the API allocates and runs your simulation tasks on computational resources that 
+are either **shared** among you and other users, or **dedicated** only to your use.
+
+- [Storage and Data Flow](./data_flow.md): Get to know the typical flow of data
```
Comment (sarmento) [inline docs/explore_api/how_it_works.md:37]: get to know --> get to understand ?

File: docs/explore_api/how_it_works.md [hunk 2]
```diff
@@ -26,23 +26,30 @@ nuances and additional options available, which we'll explore in detail througho
...
+look under the hood to learn the underlying infrastructure powering your simulations, 
+especially how the API enables you to access a variety of computational resources, 
+manages your computational workload, and allocates it to the appropriate computational 
+resource through a unified Python code.
+
+- [Templating Engine](./templating.md): Explore how the API enables you to transform 
+fixed parameter values in your "base case" simulation configuration files into 
+programmable variables you can adjust for running large variations of your simulation.
+
+- [Configuring Simulations](./configuring-simulators.md): Explore how the API
```
Comment (sarmento) [inline docs/explore_api/how_it_works.md:51]: Not sure here is this is about simulations or about simulaTORS. I think it's about simulators, right?

Comment (maiyahershey) [reply docs/explore_api/how_it_works.md:51]: yes you're right i mixed them up

**Comment by maiyahershey** [comment]
Should we change the title  of "Configuring Simulators" into "Abstracting Open-Source Simulators" instead? I don't feel the first one captures the essence well. @sarmento @sergiomsantos but we could keep this till later

**Comment by sarmento** [comment]
Let's chat about this. Maybe we could transform this explanation in a blog post. For now let's not make any substantial change in the structure. 

Let's meet about next steps...

**Comment by maiyahershey** [comment]
> Let's chat about this. Maybe we could transform this explanation in a blog post. For now let's not make any substantial change in the structure.
> 
> Let's meet about next steps...

Deal!

**Comment by sergiomsantos** [comment]
> Should we change the title of "Configuring Simulators" into "Abstracting Open-Source Simulators" instead? I don't feel the first one captures the essence well. @sarmento @sergiomsantos but we could keep this till later

Not sure "Abstracting Open-Source Simulators" is actually what we're doing there. We are actually configuring how they run.

**Review by sergiomsantos — APPROVED** [review]

**Pull request merged**